public/app/features/dashboard-scene/sharing/ShareButton/share-externally/ShareExternally.tsx

```diff
@@ -0,0 +1,186 @@
+      shareType: options[0],
+    });
+  }
+
+  onChangeShareType = (type: SelectableValue<PublicDashboardShareType>) => {
+    this.setState({ shareType: type });
+  };
+}
+
+function ShareExternallyDrawerRenderer({ model }: SceneComponentProps<ShareExternally>) {
+  const { dashboardRef, shareType, options } = model.useState();
+  const dashboard = dashboardRef.resolve();
+  const { data: publicDashboard, isLoading } = useGetPublicDashboardQuery(dashboard.state.uid!);
+
+  useEffect(() => {
+    if (publicDashboard) {
+      const opt = options.find((opt) => opt.value === publicDashboard?.share)!;
+      model.onChangeShareType(opt);
+    }
+  }, [publicDashboard, options, model]);
+
+  const hasWritePermissions = contextSrv.hasPermission(AccessControlAction.DashboardsPublicWrite);
+
+  const onCancel = useCallback(() => {
+    dashboard.closeModal();
+  }, [dashboard]);
+
+  const Config = useMemo(() => {
+    if (shareType.value === PublicDashboardShareType.EMAIL && hasEmailSharingEnabled) {
+      return <EmailSharing dashboard={dashboard} onCancel={onCancel} />;
+    }
+    if (shareType.value === PublicDashboardShareType.PUBLIC) {
+      return <PublicSharing dashboard={dashboard} onCancel={onCancel} />;
+    }
+    return <></>;
+  }, [shareType, dashboard, onCancel]);
+
+  if (isLoading) {
+    return <Loader />;
+  }
+
+  return (
+    <Stack direction="column" gap={2} data-testid={selectors.container}>
+      <ShareAlerts dashboard={dashboard} />
+      <ShareTypeSelect
+        dashboard={dashboard}
+        setShareType={model.onChangeShareType}
+        value={shareType}
+        options={options}
+      />
+      {!hasWritePermissions && <NoUpsertPermissionsAlert mode={publicDashboard ? 'edit' : 'create'} />}
+      {Config}
+      {publicDashboard && (
+        <>
+          <Divider spacing={0} />
+          <Actions dashboard={dashboard} publicDashboard={publicDashboard} />
+        </>
+      )}
+    </Stack>
+  );
+}
+
+function Actions({ dashboard, publicDashboard }: { dashboard: DashboardScene; publicDashboard: PublicDashboard }) {
+  const [update, { isLoading: isUpdateLoading }] = useUpdatePublicDashboardMutation();
+  const [deletePublicDashboard, { isLoading: isDeleteLoading }] = useDeletePublicDashboardMutation();
+
+  const isLoading = isUpdateLoading || isDeleteLoading;
+  const hasWritePermissions = contextSrv.hasPermission(AccessControlAction.DashboardsPublicWrite);
+
+  function onCopyURL() {
+    DashboardInteractions.publicDashboardUrlCopied();
+  }
+
+  const onPauseOrResumeClick = async () => {
+    DashboardInteractions.publicDashboardPauseSharingClicked({
+      paused: !publicDashboard.isEnabled,
+    });
+    update({
+      dashboard: dashboard,
+      payload: {
+        ...publicDashboard!,
+        isEnabled: !publicDashboard.isEnabled,
+      },
+    });
+  };
+
+  const onDeleteClick = () => {
+    DashboardInteractions.revokePublicDashboardClicked();
+    deletePublicDashboard({
+      dashboard,
```

Comment:
Do you need to pass dashboard to RTK queries? I would expect the RTK queries to perform server comms only, while the side effects are handled externally. Otherwise those queries can't be tested in isolation.

I would expect these kinds of functions to look like this:

```ts
const onDeleteClick = async () => {
    DashboardInteractions.revokePublicDashboardClicked();
    await deletePublicDashboard({ 
        uid: publicDashboard!.uid,
        dashboardUid: dashboard.state.uid!,
    });
   
    dashboard.onNAME_OF THE_SIDE_EFFECT(...)
}
```

Reply:
yes, we can review this. I'll add it as part of our tech debt epic. It's out of the scope of this PR

Reply:
Added [here](https://github.com/grafana/grafana/issues/89118)